org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/xsd/contentmodel/CMXSDDocument.java

```diff
@@ -520,6 +519,18 @@ private static LocationLink findXSElement(DOMElement originElement, NodeList chi
 		return null;
 	}
 
+	private static LocationLink findXSElement(DOMElement originElement, Element elt) {
+		if (elt != null && XSDUtils.isXSElement(elt)) {
+			if (originElement.getLocalName().equals(elt.getAttribute("name"))) {

```

Comment:
can getLocalName() return null?

Reply:
fixed, I check that before in the start of findTypeLocation